pkg/registry/common/expire/nse_server.go

```diff
@@ -21,31 +21,37 @@ import (
 	"sync"
 	"time"
 
+	"github.com/networkservicemesh/sdk/pkg/registry/core/next"
+	"github.com/networkservicemesh/sdk/pkg/tools/serialize"
+
 	"github.com/golang/protobuf/ptypes/empty"
 	"github.com/networkservicemesh/api/pkg/api/registry"
 )
 
 type nseServer struct {
-	nses      map[string]*registry.NetworkServiceEndpoint
-	nsesMutex sync.Mutex
-	period    time.Duration
-	once      sync.Once
-	server    registry.NetworkServiceEndpointRegistryServer
+	nses     map[string]*registry.NetworkServiceEndpoint
+	executor serialize.Executor
+	period   time.Duration
+	once     sync.Once
+	server   registry.NetworkServiceEndpointRegistryServer
 }
 
 func (n *nseServer) setPeriod(d time.Duration) {
 	n.period = d
 }
 
 func (n *nseServer) Register(ctx context.Context, nse *registry.NetworkServiceEndpoint) (*registry.NetworkServiceEndpoint, error) {
-	n.once.Do(n.monitor)
-	r, err := n.server.Register(ctx, nse)
+	n.once.Do(func() {
+		n.server = next.NetworkServiceEndpointRegistryServer(ctx)
```

Comment:
Super curious why this is in the once.Do ... I think its probably harmless... but curious :)

Reply:
It is needed to call unregister for the next chain elements when NSE/NS has achieved expiration time.

Reply:
Why not just call Unregister against expire itself?

Reply:
Should be fixed.